docs/security-policy-for-authors.md

```diff
@@ -0,0 +1,181 @@
+# Security Policy
+
+Ideally, there would be a separate `SECURITY` `SECURITY.md` or `SECURITY.pod` file in
+the distribution that outlines the security policy in more detail.
+
+Note that `SECURITY.txt` or `security.txt` is discouraged, as that is
+often associated with https://securitytxt.org/ which has a different purpose.
+
+A minimal security policy should discuss
+
+1. How to report a security vulnerability
+
+   At the bare minimum, it should provide a current package
+   maintainer's email address (or ideally an alias that forwards to
+   multiple authors), or a bug reporting system that supports special
+   handling of security issues.
+
+   Some examples,
+
+   > Security vulnerabilties can be reported by e-mail to the current
+   > project maintainers at <fobar@example.com>.
```

Comment:
Note that it is important to refer to the "current maintainers" vs the "authors" as they may be different, especially for older modules.

Reply:
Yeah, this is a really good point. Strictly speaking, looks like a "depends on the situation" case, so it may be worth find some neutral language here also.

In the CRA [Annex I, part 2 (6)](https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=OJ:L_202402847#page=68), the language is

> Manufacturers of products with digital elements shall:
> (2) take measures to facilitate the sharing of information about potential vulnerabilities in their product with digital elements as well as in third-party components contained in that product, including by **providing a contact address for the reporting of the vulnerabilities discovered in the product** with digital elements

…and in [Annex II](https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=OJ:L_202402847#page=70), the language is

> At minimum, the product with digital elements shall be accompanied by:
> (2) the **single point of contact** where information about vulnerabilities of the product with digital elements can be reported and received, and where the manufacturer’s policy on coordinated vulnerability disclosure can be found

This is relevant due to Article 13 (6), which basically requires manufacturers to report found vulnerabilities and assist maintainers with any fixes they have made.

> Manufacturers shall, upon identifying a vulnerability in a component, including in an open source-component, which is integrated in the product with digital elements report the vulnerability to the person or entity manufacturing or maintaining the component, and address and remediate the vulnerability in accordance with the vulnerability handling requirements set out in Part II of Annex I.
> Where manufacturers have developed a software or hardware modification to address the vulnerability in that component, they shall share the relevant code or documentation with the person or entity manufacturing or maintaining the component, where appropriate in a machine-readable format.

This may also have some relevance due to [Article 24.1](https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=OJ:L_202402847#page=45) that says something about fostering effective handling of vulnerabilities.

> 1. Open-source software stewards shall put in place and document in a verifiable manner a cybersecurity policy to foster the development of a secure product with digital elements as well as an effective handling of vulnerabilities by the developers of that product. That policy shall also foster the voluntary reporting of vulnerabilities as laid down in Article 15 by the developers of that product and take into account the specific nature of the open-source software steward and the legal and organisational arrangements to which it is subject. That policy shall, in particular, include aspects related to documenting, addressing and remediating vulnerabilities and promote the sharing of information concerning discovered vulnerabilities within the open-source community.

How about using the phrase "single point of contact for reporting vulnerabilities"?

Reply:
Done, but "single point of contact" seems to conflict with the copy/forward to CPAN Security Group if 0-day or you've not heard back.

It makes me think that cpan.org should provide small distribution-list emails for security that forwards to the package maintainers *and* the security group.

Reply:
One point of clarification: The "single point of contact" requirement is for _manufacturers_ mainly, and for any open source projects who decide to "opt in" by deciding their project is "intended for commercial use" and select an Open Source Steward to help them comply with the "light touch" requirements in the CRA.

This means that having a "single point of contact" is _still_ voluntary and optional. We'll need to recommend things that works both when maintainers choose to "opt in" and those who don't care.

Suggestion: Recommend that maintainers publish a "single point of contact", but if they don't then use any other contact methods mentioned, or as a last resort - contact CPANSec.

Reply:
Apologies for the tangent. Feel free to ignore my public thinking; your text seems fine. :)